logstreams/src/test/java/io/zeebe/distributedlog/DistributedLogTest.java

```diff
@@ -105,6 +112,52 @@ public void shouldReplicateMultipleEvents()
     assertEventsCount(node3, 3);
   }
 
+  @Test
+  public void shouldReplicateMultipleEventsWithFailure()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    // given
+    final CountDownLatch latch = new CountDownLatch(1);
+    final var failingFuture = new CompletableFuture<Long>();
+    node1.waitUntilNodesJoined();
+    node2.waitUntilNodesJoined();
+    node3.waitUntilNodesJoined();
+
+    node1.becomeLeader(PARTITION_ID);
+
+    doReturn(failingFuture)
+        .doCallRealMethod()
+        .when(node1.getSpy(PARTITION_ID))
+        .asyncAppend(eq(2L), any(byte[].class), anyLong());
+
+    doAnswer(
+            i -> {
+              latch.countDown();
+              return i.callRealMethod();
+            })
+        .doCallRealMethod()
+        .when(node1.getSpy(PARTITION_ID))
+        .asyncAppend(eq(3L), any(byte[].class), anyLong());
+
+    // when
+    final Event event1 = writeEvent("record1");
+    Thread.sleep(100L); // to get events in different chunks
```

Comment:
Instead of sleep, can we do something like 
`verify(node1.getSpy(PARTITION_ID), timeout(1000).times(1)).asyncAppend(1L, any(), any());`

Reply:
This wouldnt split the blocks, since the appender tries to consume as many as possible from the dispatcher